src/lib/connection.ts

```diff
@@ -70,6 +70,25 @@ export interface MethodOptions {
   timeout?: number;
 }
 
+export interface CreateOptions {
```

Comment:
the `MethodOptions` interface was used for all of these, which was confusing at best.  Splitting these into their own interfaces seems to make more sense and be better/easier for typescript devs